tfe/data_source_oauth_clients_test.go

```diff
@@ -0,0 +1,68 @@
+package tfe
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccTFEOAuthClientsDataSource_basic(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			if GITHUB_TOKEN == "" {
+				t.Skip("Please set GITHUB_TOKEN to run this test")
+			}
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{Destroy: false,
+				PreventPostDestroyRefresh: true,
+				Config:                    testAccTFEOAuthClientsDataSourceConfig(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.tfe_oauth_clients.foobar", "oauth_clients.#", "2"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_oauth_clients.foobar", "oauth_clients.0.service_provider", "github"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_oauth_clients.foobar", "oauth_clients.1.service_provider", "github"),
+				),
+			},
+		},
+	})
+}
+
+func testAccTFEOAuthClientsDataSourceConfig(rInt int) string {
```

Comment:
I would also write a test/config for when there are no oauth clients to fetch. Tests should capture multiple configuration scenarios other than 'basic' (even though this DS is quite simple), the reason being users have a knack for breaking things! 😆